clients/drcachesim/tracer/tracer.cpp

```diff
@@ -513,8 +521,12 @@ append_marker_seg_base(void *drcontext, func_trace_entry_vector_t *vec)
      * a redzone check at the end guarding a clean call to memtrace(), but to
      * be a litte safer in case that changes we also do a redzone check here.
      */
-    if (BUF_PTR(data->seg_base) - data->buf_base > static_cast<ssize_t>(trace_buf_size))
+    if (BUF_PTR(data->seg_base) - data->buf_base > static_cast<ssize_t>(trace_buf_size)) {
+        BUF_PTR(data->seg_base) += instru->append_timestamp(BUF_PTR(data->seg_base));
```

Comment:
Maybe create a separate method that does append_timestamp -> append_marker -> process_and_output_buffer?

Reply:
Just the 1st 2 steps are called 5 times, while only 2 of those also call process_and_output.  So I made a helper for the 1st 2 steps, the 2 appends.